src/mame/heathkit/intr_cntrl.h

```diff
@@ -73,4 +92,51 @@ class z37_intr_cntrl : public heath_intr_cntrl
 DECLARE_DEVICE_TYPE(HEATH_INTR_CNTRL, heath_intr_cntrl)
 DECLARE_DEVICE_TYPE(HEATH_Z37_INTR_CNTRL, z37_intr_cntrl)
 
+
+class heath_intr_socket : public device_t,
+							public device_single_card_slot_interface<device_heath_intr_interface>
+{
+public:
+
+	template <typename T>
+	heath_intr_socket(const machine_config &mconfig, const char *tag, device_t *owner, T &&opts, const char *dflt, bool fixed = false) :
+		 heath_intr_socket(mconfig, tag, owner, 0)
+	{
+		option_reset();
+		opts(*this);
+		set_default_option(dflt);
+		set_fixed(fixed);
+	}
+
+	heath_intr_socket(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock = 0);
+	virtual ~heath_intr_socket();
+
+	// required operation overrides
+	virtual void set_irq_level(uint8_t level, int state) { if (m_cntrl) { m_cntrl->set_irq_level(level, state); }};
+
+	IRQ_CALLBACK_MEMBER(irq_callback);
+
+	auto irq_line_cb() { return m_irq_line.bind(); }
+	virtual uint8_t get_instruction();
+
+	void raise_irq(int state) { m_irq_line(state); }
+
+	void set_irq(int state) { if (m_cntrl) { m_cntrl->set_irq(state); }};
+	void set_drq(int state) { if (m_cntrl) { m_cntrl->set_drq(state); }};
+	void block_interrupts(uint8_t data) { if (m_cntrl) { m_cntrl->block_interrupts(data); }};
```

Comment:
Is there a reason `set_irq_level` and `raise_irq` are virtual?  They shouldn’t need to be since they’re just passed on to the connected device.